fe/fe-core/src/main/java/org/apache/doris/analysis/IndexDef.java

```diff
@@ -225,6 +225,9 @@ public void checkColumn(Column column, KeysType keysType, boolean enableUniqueKe
                 throw new AnalysisException(colType + " is not supported in " + indexType.toString() + " index. "
                         + "invalid index: " + indexName);
             }
+            if (colType.isVariantType() && isIndexFormatV1) {
```

Comment:
add comment why:
1. performance, too many files and small io
2. some link file issue

Reply:
done